### DISCUSSION TITLE
How can I leave 'as is' data in nested array object?

### DISCUSSION THREAD

**Discussion by n-shkutov-awtg** [react-hook-form/react-hook-form#3156] [Q&A] [answered]
I fetch the `data` from server like this:
```
{
  id: 0,
  result: true
}
```
Then I use `reset(data)` to put this into form state, then I use one Selector (React-Select + Controller) with `someNestedKey[`${index}`].result` name, then  I change value and it change `result` inside of initial or reseted object and also it affected shape of the object. It becomes just :
```
{
  result: true
}
```

How can I save `id: 0` inside of nested array object?

**Comment by bluebill1049** [comment, marked as answer]
keyName: '_id' to avoid id name collision?
